### PR TITLE
Bkit 162 user repository

### PR DESCRIPTION
**Pull request by mathomas**
Only the bits needed to save Users into a USER table.  Backed out anything having to do with Booking/User association.

### CODE REVIEW

File: src/main/resources/schema.sql [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
 DROP TABLE IF EXISTS BOOKABLE;
 DROP TABLE IF EXISTS LOCATION;
 DROP TABLE IF EXISTS BOOKING;
 
+CREATE TABLE USER (
+    USER_ID            VARCHAR(36) NOT NULL,
+    CONSTRAINT PK_USER PRIMARY KEY (USER_ID),
+    EXTERNAL_USER_ID   VARCHAR(36) NOT NULL,
+    GIVEN_NAME         VARCHAR(128) NOT NULL,
+    FAMILY_NAME        VARCHAR(128) NOT NULL
```
Comment (aochsner) [inline src/main/resources/schema.sql]: NOT NULL on names might be making an incorrect assumption...

File: src/main/resources/schema.sql [hunk 2]
```diff
@@ -11,17 +22,19 @@ CREATE TABLE BOOKABLE (
 );
 
 CREATE TABLE LOCATION (
-    LOCATION_ID   VARCHAR(36) NOT NULL,
-    LOCATION_NAME VARCHAR(50) NOT NULL,
-    LOCATION_TZ   VARCHAR(50) NOT NULL
+    LOCATION_ID        VARCHAR(36) NOT NULL,
```
Comment (aochsner) [inline src/main/resources/schema.sql]: add primary key whilst we're here?

File: src/main/resources/schema.sql [hunk 3]
```diff
@@ -1,7 +1,18 @@
+DROP TABLE IF EXISTS USER;
 DROP TABLE IF EXISTS BOOKABLE;
 DROP TABLE IF EXISTS LOCATION;
 DROP TABLE IF EXISTS BOOKING;
 
+CREATE TABLE USER (
+    USER_ID            VARCHAR(36) NOT NULL,
+    CONSTRAINT PK_USER PRIMARY KEY (USER_ID),
+    EXTERNAL_USER_ID   VARCHAR(36) NOT NULL,
```
Comment (aochsner) [inline src/main/resources/schema.sql]: I'd bump this to 64 at least (sha-256 length)... not all providers use GUID like microsoft

File: src/main/resources/schema.sql (hunk 2 above)
Comment (mathomas) [reply src/main/resources/schema.sql]: Heh.  Was trying to keep the blast radius related directly to the story/task, but yeah.

**Pull request merged**